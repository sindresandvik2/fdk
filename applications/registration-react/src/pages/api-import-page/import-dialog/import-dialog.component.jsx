import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'reactstrap';

import localization from '../../../utils/localization';
import { PanelBase } from '../../../components/panel-base/panel-base.component';
import { InputFile } from '../../../components/input-file/input-file.component';

export const ImportDialog = props => {
  const { showLinkImport, handleFileUpload } = props;
  return (
    <PanelBase>
      <div className="mb-5">
        <h3>{localization.api.import.importApiSpec}</h3>
      </div>

      <Button className="mr-3" color="primary" onClick={showLinkImport}>
        {localization.api.import.importLink}
      </Button>
      <InputFile className="btn btn-primary m-0" onChange={handleFileUpload}>
        {localization.api.import.importFile}
      </InputFile>
    </PanelBase>
  );
};

ImportDialog.propTypes = {
  showLinkImport: PropTypes.func.isRequired,
  handleFileUpload: PropTypes.func.isRequired
};
