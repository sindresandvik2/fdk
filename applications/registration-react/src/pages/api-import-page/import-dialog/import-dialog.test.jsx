import React from 'react';
import { shallow } from 'enzyme';
import { ImportDialog } from './import-dialog.component';

let defaultProps;
let wrapper;
let preventDefault;
let showLinkImport;
let handleFileUpload;

beforeEach(() => {
  showLinkImport = jest.fn();
  handleFileUpload = jest.fn();
  preventDefault = jest.fn();
  defaultProps = {
    showLinkImport,
    handleFileUpload
  };
  wrapper = shallow(<ImportDialog {...defaultProps} />);
});

test('should render ImportDialog correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle button show link upload', () => {
  const mockedEvent = {
    preventDefault
  };
  wrapper.find('Button').prop('onClick')(mockedEvent);
  expect(showLinkImport).toHaveBeenCalled();
});

test('should handle input file upload', () => {
  const mockedEvent = {
    preventDefault
  };
  wrapper.find('InputFile').prop('onChange')(mockedEvent);
  expect(handleFileUpload).toHaveBeenCalled();
});
