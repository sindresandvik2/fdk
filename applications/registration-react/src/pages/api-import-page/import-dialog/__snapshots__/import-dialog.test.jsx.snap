// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ImportDialog correctly 1`] = `
<PanelBase>
  <div
    className="mb-5"
  >
    <h3>
      Importer API-spesifikasjon
    </h3>
  </div>
  <Button
    className="mr-3"
    color="primary"
    onClick={[MockFunction]}
    tag="button"
  >
    Importer via lenke
  </Button>
  <InputFile
    className="btn btn-primary m-0"
    onChange={[MockFunction]}
  >
    Importer via fil
  </InputFile>
</PanelBase>
`;
